File: examples/fib.lua
local function fib(n)
    return n < 2 and n or fib(n - 1) + fib(n - 2)
end

local function print_terms(i, n)
    if i < n then
        print(fib(i))
        print_terms(i + 1, n)
    end
end

local function main()
    print_terms(0, 35)
end

main()

File: examples/fib.py
def fib(n: int) -> int:
    return n if n < 2 else fib(n - 1) + fib(n - 2)

def print_terms(i: int, n: int) -> None:
    if i < n:
        print(fib(i))
        print_terms(i + 1, n)

def main() -> None:
    print_terms(0, 35)

if __name__ == "__main__":
    main()
